app/src/test/java/de/westnordost/streetcomplete/quests/opening_hours/AddOpeningHoursTest.kt

```diff
@@ -47,6 +48,40 @@ class AddOpeningHoursTest {
         )
     }
 
+    @Test fun `apply description answer when it was not signed before`() {
+        questType.verifyAnswer(
+            mapOf("opening_hours:signed" to "no"),
+            DescribeOpeningHours("my cool \"opening\" hours"),
+            StringMapEntryAdd("opening_hours", "\"my cool opening hours\""),
+            StringMapEntryDelete("opening_hours:signed", "no")
+        )
+    }
+
+    @Test fun `apply description answer when it was not signed but already had an opening hours`() {
+        questType.verifyAnswer(
+            mapOf(
+                "opening_hours" to "my opening hours",
+                "opening_hours:signed" to "no"
+            ),
+            DescribeOpeningHours("my cool \"opening\" hours"),
+            StringMapEntryModify("opening_hours", "my opening hours", "\"my cool opening hours\""),
+            StringMapEntryDelete("opening_hours:signed", "no")
+        )
+    }
+
+    @Test fun `apply same description answer again when it was not signed before`() {
+        questType.verifyAnswer(
+            mapOf(
+                "opening_hours" to "\"oh\"",
+                "opening_hours:signed" to "no"
+            ),
+            DescribeOpeningHours("oh"),
+            StringMapEntryModify("opening_hours", "\"oh\"", "\"oh\""),
+            StringMapEntryAdd("check_date:opening_hours", LocalDate.now().toCheckDateString()),
```

Comment:
It feels to me like technically it shouldn't tag the check_date here given other fields have changed (given that's how SC operates in general now), but I appreciated that's all a bit complicated when it's a related field that's changed.

Reply:
yeah